### PR TITLE
added edit mode

### LINKED ISSUE #3857: Expenditures: As a user I can edit streaming payment
[Provided by Adrian via Contractors project management platform]

## Description
User has to be able to edit streaming payment. They should be able to edit all information in settings and funding source form.

## Acceptance Criteria

- forms should look like in an initial state
- an option to see a label "You are editing this payment" above settings
- an option to click "Cancel" button
- an option to click "Update" button

## Design

- [Figma Link - Edit mode of streaming](https://www.figma.com/file/dCtvv76S8mk5HNApjwmBto/Expenditures?node-id=5731%3A324164)

Until - popover 9

### PR DESCRIPTION
**Pull request by agnieszkajarosikloj**
## Description

This PR adds edition in locked state in streaming payment. 
When streaming payment is in locked state, user sees an 'edit' icon. After clicking on it user can change payment details (settings and funding source). 
The modal doesn't show user changes yet. Modal will be added in another PR.  Necessary changes have not yet been merged.
Settings section in locked state is temporary. Now it looks like in other payments type. After merging this [PR](https://github.com/JoinColony/colonyDapp/pull/3867) it will be replaced with the correct one.
Screenshot 2022-09-23 at 16 09 03

Resolves #3857

### CODE REVIEW

**Review by jakubcolony — COMMENTED** [review]
Looks good to me!
image

Left a few comments/questions

File: src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ import StreamingStages from './StreamingStages';
 const displayName = 'dashboard.ExpenditurePage.Stages.FormStages';
 
 interface Props {
-  states: State[];
+  stages: StageObject[];
   activeStateId?: string;
```
Comment (jakubcolony) [inline src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.tsx:22]: A few occurrences of the old `state` name

File: src/modules/dashboard/components/ExpenditurePage/Stages/LockedStages.tsx [hunk 2]
```diff
@@ -44,7 +45,7 @@ interface Props {
 }
 
 const LockedStages = ({
-  states,
+  stages,
   activeStateId,
```
Comment (jakubcolony) [inline src/modules/dashboard/components/ExpenditurePage/Stages/LockedStages.tsx:49]: A few occurrences of the old `state` name

File: src/modules/dashboard/components/ExpenditurePage/Stages/Stages.tsx [hunk 3]
```diff
@@ -83,7 +83,7 @@ export interface Props {
 }
 
 const Stages = ({
-  states,
+  stages,
   activeStateId,
```
Comment (jakubcolony) [inline src/modules/dashboard/components/ExpenditurePage/Stages/Stages.tsx:87]: A few occurrences of the old `state` name

**Review by jakubcolony — CHANGES_REQUESTED** [review]
It looks good @agnieszkajarosikloj! I noticed you renamed `state` to `stage` so I highlighted a few places where it's still using the old name. Would be great to change it all for consistency.

**Review by jakubcolony — APPROVED** [review]
Great work!

**Review by arrenv — APPROVED** [review]
Looks good to me @agnieszkajarosikloj

**Review by chinins — APPROVED** [review]
Very good!

**Pull request merged**